streams/src/test/java/org/apache/kafka/streams/integration/StreamTableJoinIntegrationTest.java

```diff
@@ -56,6 +60,32 @@ public void prepareTopology() throws InterruptedException {
         leftStream = builder.stream(INPUT_TOPIC_LEFT);
     }
 
+    @Test
+    public void testShouldAutoShutdownOnIncompleteMetadata() throws InterruptedException {
+        STREAMS_CONFIG.put(StreamsConfig.APPLICATION_ID_CONFIG, appID + "-incomplete");
+
+        KStream<Long, String> notExistStream = builder.stream(INPUT_TOPIC_LEFT + "-not-existed");
+
+        KTable<Long, String> aggregatedTable = notExistStream.leftJoin(rightTable, valueJoiner)
+                .groupBy((key, value) -> key)
+                .reduce((value1, value2) -> value1 + value2);
+
+        // Write the (continuously updating) results to the output topic.
+        aggregatedTable.toStream().to(OUTPUT_TOPIC);
+
+        final KafkaStreamsWrapper streams = new KafkaStreamsWrapper(builder.build(), STREAMS_CONFIG);
+        final IntegrationTestUtils.StateListenerStub listener = new IntegrationTestUtils.StateListenerStub();
+        streams.setStreamThreadStateListener(listener);
+        streams.start();
+
+        while (!listener.revokedToPendingShutdownSeen()) {
```

Comment:
maybe use `TestUtils.waitForCondition` here while not probable this could hang.

Reply:
Good point.